Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp

```diff
@@ -594,9 +594,14 @@ void WebFrameLoaderClient::dispatchDidCommitLoad(std::optional<HasInsecureConten
     bool wasPrivateRelayed = documentLoader.response().wasPrivateRelayed();
     if (!wasPrivateRelayed && wasPrivateRelayedFromPageCache)
         wasPrivateRelayed = wasPrivateRelayedFromPageCache == WasPrivateRelayed::Yes;
-    
+
+    auto certificateInfo = valueOrCompute(documentLoader.response().certificateInfo(), [] {
+        return CertificateInfo();
+    });
+    hasInsecureContent = hasInsecureContent ? *hasInsecureContent : (certificateInfo.containsNonRootSHA1SignedCertificate() ? HasInsecureContent::Yes : HasInsecureContent::No);
```

Comment:
We should probably check with @pvollan that we're OK doing the SecTrustEvaluate calls from the WebProcess in the long term (because of sandboxing reasons).

Reply:
I did.  I verified there's already a call to containsNonRootSHA1SignedCertificate in the web process and it hasn't caused any problems.  I also verified that this doesn't cause any issues.  I think it's because we call response.includeCertificateInfo in NetworkLoad::notifyDidReceiveResponse before sending the SecTrustRef to the web process, so the web process doesn't actually need to do anything that our sandbox would oppose.